bindings/wasm/src/credential/domain_linkage_validator.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::Result;
+use crate::error::WasmResult;
+use crate::resolver::RustSupportedDocument;
+use identity_iota::core::Url;
+use identity_iota::credential::Credential;
+use identity_iota::credential::CredentialValidationOptions;
+use identity_iota::credential::DomainLinkageConfiguration;
+use identity_iota::credential::DomainLinkageValidator;
+use proc_typescript::typescript;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+/// A validator for a Domain Linkage Configuration and Credentials.
+#[wasm_bindgen(js_name = DomainLinkageValidator)]
+pub struct WasmDomainLinkageValidator;
+
+#[wasm_bindgen(js_class = DomainLinkageValidator)]
+impl WasmDomainLinkageValidator {
+  /// Validates the linkage between a domain and a DID.
+  /// [`DomainLinkageConfiguration`] is validated according to [DID Configuration Resource Verification](https://identity.foundation/.well-known/resources/did-configuration/#did-configuration-resource-verification).
+  ///
+  /// Linkage is valid if no error is thrown.
+  ///
+  /// # Note:
+  /// - Only [Linked Data Proof Format](https://identity.foundation/.well-known/resources/did-configuration/#linked-data-proof-format)
+  ///   is supported.
+  /// - Only the Credential issued by `issuer` is verified.
+  ///
+  /// # Errors
+  ///  - Semantic structure of `configuration` is invalid.
+  ///  - `configuration` includes multiple credentials issued by `issuer`.
+  ///  - Validation of the matched Domain Linkage Credential fails.
+  #[wasm_bindgen(js_name = validateLinkage)]
+  pub fn validate_linkage(options: IValidateLinkage) -> Result<()> {
+    let IValidateLinkageHelper {
+      issuer,
+      configuration,
+      domain,
+      validation_options,
+    } = options.into_serde::<IValidateLinkageHelper>().wasm_result()?;
+
+    match issuer {
+      RustSupportedDocument::Core(doc) => {
+        DomainLinkageValidator::validate_linkage(&doc, &configuration, &domain, &validation_options).wasm_result()?;
+      }
+      RustSupportedDocument::Iota(doc) => {
+        DomainLinkageValidator::validate_linkage(&doc, &configuration, &domain, &validation_options).wasm_result()?;
+      }
+    }
+    Ok(())
+  }
+
+  /// Validates a [Domain Linkage Credential](https://identity.foundation/.well-known/resources/did-configuration/#domain-linkage-credential).
+  /// Error will be thrown in case the validation fails.
+  #[wasm_bindgen(js_name = validateCredential)]
+  pub fn validate_credential(options: IValidateCredential) -> Result<()> {
+    let IValidateCredentialHelper {
+      issuer,
+      credential,
+      domain,
+      validation_options,
+    } = options.into_serde::<IValidateCredentialHelper>().wasm_result()?;
+
+    match issuer {
+      RustSupportedDocument::Core(doc) => {
+        DomainLinkageValidator::validate_credential(&doc, &credential, domain, &validation_options).wasm_result()?;
```

Comment:
There's an inconsistency here between `validate_credential` taking `Url` and `validate_linkage` taking `&Url`. As far as I can tell both can take `&Url` which saves you some clones.